opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/reminder/SyncAccountAlarmListener.java

```diff
@@ -0,0 +1,33 @@
+package de.geeksfactory.opacclient.reminder;
+
+import com.commonsware.cwac.wakeful.WakefulIntentService;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+
+public class SyncAccountAlarmListener implements WakefulIntentService.AlarmListener {
+    // We could use a longer interval like 3 hours, but on API < 19, only the AlarmManager
+    // .INTERVAL_... constants make the AlarmManager use inexact alarm delivery for better
+    // battery efficiency. The next larger constant available would be AlarmManager
+    // .INTERVAL_HALF_DAY.
```

Comment:
> We only need a way for the scheduled run to determine if there already is an hourly run scheduled and only reschedule itself on failure if there isn't, right?

We could save the current interval (either 1 hour or half a day) in the `SharedPreferences` and let the `SyncAccountAlarmListener.scheduleAlarms` function use this to schedule alarms in this interval. There, `firstStart` would also need to be set to the current time plus the interval.

Then, every time the `SyncAccountService` runs, it will update the preference value depending on if it failed or not and reschedule the Alarms by calling `WakefulIntentService.cancelAlarms` and `WakefulIntentService.scheduleAlarms`. If the service does not get to this point for whatever reason, the alarms will stay as they were, so it will still be called again the next time.